tests/integrationv2/providers.py

```diff
@@ -225,22 +224,24 @@ def _join_ciphers(self, ciphers):
     def _cipher_to_cmdline(self, protocol, cipher):
         cmdline = list()
 
-        if cipher.min_version is Protocols.TLS13:
-            cmdline.append('-ciphersuites')
-        else:
-            cmdline.append('-cipher')
-
         ciphers = []
         if type(cipher) is list:
             ciphers.append(self._join_ciphers(cipher))
+            min_version = min(cipher, key=lambda x: x.min_version)
         else:
+            min_version = cipher.min_version
             if cipher.min_version == Protocols.TLS13:
                 ciphers.append(OpenSSL.supported_ciphers[cipher])
             else:
                 # This replace is only done for ciphers that are not TLS13 specific.
                 # Run `openssl ciphers` to view the inconsistency in naming.
                 ciphers.append(cipher.name.replace("_", "-"))
 
+        if min_version is Protocols.TLS13:
+            cmdline.append('-ciphersuites')
+        else:
+            cmdline.append('-cipher')
```

Comment:
Does this work with a mixed list of TLS1.2 and TLS1.3 ciphers?

Reply:
I went back to double check this.

-ciphersuites must contain only TLS1_3 ciphers
-cipher can be mixed, but will not negotiate the TLS1_3 ciphers. The protocol is still negotiated as TLS1_3 though.

In our version negotiation test, we are only verifying the negotiated version. So that test passes as expected. We are only using a single cipher as well, so this mixed cipher list isn't covered by the version negotiation test.

In the SNI match test we have multiple accepted ciphers, but we don't cross the TLS1_2 / TLS1_3 boundary. So there is nothing explicitly testing which cipher was chosen in a list of OpenSSL ciphers.

I think this warrants a comment to describe the reason we care about the check. I'm not sure what behavior an additional integ test should cover here. It seems like we would be checking the behavior of OpenSSL instead of S2N.